examples/md-flexible/src/configuration/objects/CubeGauss.h

```diff
@@ -92,22 +94,18 @@ class CubeGauss : public Object {
    * @param particles The container where the new particles will be stored.
    */
   void generate(std::vector<ParticleType> &particles) const override {
-    ParticleType particle = getDummyParticle(particles.size());
-
-    std::default_random_engine generator(42);
-    std::array<std::normal_distribution<double>, 3> distributions = {
-        std::normal_distribution<double>{_distributionMean[0], _distributionStdDev[0]},
-        std::normal_distribution<double>{_distributionMean[1], _distributionStdDev[1]},
-        std::normal_distribution<double>{_distributionMean[2], _distributionStdDev[2]}};
-
-    for (int i = 0; i < _numParticles; ++i) {
-      particle.setR({_bottomLeftCorner[0] + distributions[0](generator),
-                     _bottomLeftCorner[1] + distributions[1](generator),
-                     _bottomLeftCorner[2] + distributions[2](generator)});
-
-      particles.push_back(particle);
-      particle.setID(particle.getID() + 1);
-    }
+    // Wrapper so that std::vector can be used as an AutoPas::ParticleContainer
+    auto particlesWrapper = autopasTools::PseudoContainer(particles);
+
+    using namespace autopas::utils::ArrayMath::literals;
+    const auto boxMax = _bottomLeftCorner + _boxLength;
+
+    // dummy particle used as a template with id of the first newly generated one
+    const ParticleType dummyParticle = getDummyParticle(particles.size());
+
+    autopasTools::generators::GaussianGenerator::fillWithParticles(particlesWrapper, _bottomLeftCorner, boxMax,
```

Comment:
This function in `GaussianGenerator` applies some error-checking:
```cpp
if (attempts == _maxAttempts) {
  std::ostringstream errormessage;
  errormessage << "GaussianGenerator::fillWithParticles(): Could not find a valid position for particle " << i
               << "after" << _maxAttempts << "attempts. Check if your parameters make sense:" << std::endl
               << "BoxMin       = " << autopas::utils::ArrayUtils::to_string(boxMin) << std::endl
               << "BoxMax       = " << autopas::utils::ArrayUtils::to_string(boxMax) << std::endl
               << "Gauss mean   = " << autopas::utils::ArrayUtils::to_string(distributionMean) << std::endl
               << "Gauss stdDev = " << autopas::utils::ArrayUtils::to_string(distributionStdDev) << std::endl;
  throw std::runtime_error(errormessage.str());
}
```

Assuming **this logic is correct**, then the `CubeGauss` object from the example `AllOptions.yaml` file with the subsequently listed properties

```yaml
CubeGauss:
  0:
    distribution-mean              :  [2, 2, 2]
    distribution-stddeviation      :  [2, 2, 2]
    numberOfParticles              :  100
    box-length                     :  [8, 8, 8]
    bottomLeftCorner               :  [15, 15, 15]
    velocity                       :  [0, 0, 0]
    particle-type-id               :  0
```

leads to the following error

```
libc++abi: terminating due to uncaught exception of type std::runtime_error: GaussianGenerator::fillWithParticles(): Could not find a valid position for particle 0after100attempts. Check if your parameters make sense:
BoxMin       = [15, 15, 15]
BoxMax       = [23, 23, 23]
Gauss mean   = [2, 2, 2]
Gauss stdDev = [2, 2, 2]
```

The check also fails with the particle number set to `1`, it only (trivially) succeeds with particle number set to `0`.
So, either
- [ ] The object specification is wrong --> Requires a change of all `example.yaml` files using the `GaussianGenerator`

or

- [ ] The checking algorithm is somehow broken --> Requires a modification of `GaussianGenerator::fillWithParticles`

Reply:
The mean is outside of the box so the generator works as intended. Will fix the example.

Reply:
Oh indeed, uff, it seems my "Brain - Department for Math Affairs" has not yet been occupied this morning.